services/121-service/run-performance-tests.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# Performance test runner script
+# This script runs performance tests sequentially, restarting services between tests
+# Similar to k6/run-all-tests.sh but for Jest
+
+# Array to collect failed tests
+failed_tests=()
+
+# Performance test files to run
+test_files=(
+  "test/performance/import-1000-registrations.test.ts"
+  "test/performance/get-program-many-attributes.test.ts"
+  "test/performance/bulk-update-32k-registrations.test.ts"
+  "test/performance/find-duplicates-100k-registrations.test.ts"
+  "test/performance/retry-failed-jobs-startup.test.ts"
+  "test/performance/performance-during-payment.test.ts"
+  "test/performance/payment-100k-intersolve-visa.test.ts"
+  "test/performance/payment-100k-safaricom.test.ts"
+  "test/performance/status-change-payment-large-program.test.ts"
+)
+
+echo "Found the following performance test files to run:"
+
+for file in "${test_files[@]}"; do
+  echo " - ${file}"
+done
+
+# new line
+echo ""
+
+for file in "${test_files[@]}"; do
+  echo "::group::Running performance test: ${file}"
+
+  test_name=$(basename "${file}" .test.ts)
+  echo "Creating log directory"
+  log_dir="${PWD}/logs/${test_name}"
+  mkdir -p "${log_dir}"
+
+  echo "Starting services"
+  (cd ../services || exit 1 ; docker --log-level 'warn' compose -f docker-compose.yml up -d --quiet-pull --wait --wait-timeout 300)
+
+  echo "Running Jest performance test"
+  # Run specific test file with integration test configuration
+  npm run test:integration:all -- --testPathPattern="${file}" --verbose
+
+  # Check exit code
+  if [[ $? -ne 0 ]]; then
```

Comment:
The exit code should be assigned as the output of the previous command, and then checked here, because any other failure in the script as a whole should also report failure.